src/ensembl/src/content/app/browser/chromosome-navigator/ChromosomeNavigator.test.tsx

```diff
@@ -0,0 +1,388 @@
+        );
+        const randomDistance = random(smallDistance);
+        const viewportStart = 20000;
+        const viewportEnd = viewportStart + randomDistance;
+
+        const props = { ...minimalProps, viewportStart, viewportEnd };
+        const { openingBracket, closingBracket } = getRenderedViewport(props);
+
+        // we expect the viewport’s start to be defined by props.viewportStart,
+        // and the viewport’s end to be two bracket widths to the right of viewport start
+        const expectedStartX = scalingFactor * props.viewportStart;
+        const expectedEndX =
+          expectedStartX + 2 * constants.VIEWPORT_BRACKET_BAR_WIDTH;
+        assertBracketX(openingBracket, expectedStartX);
+        assertBracketX(closingBracket, expectedEndX);
+      });
+
+      it('draws two viewport areas if start position is greater than end position', () => {
+        // this emulates what we may encounter in circular chromosomes,
+        // when user pans past the nominal end position of the chromosome,
+        // and sees the end of the chromosome on the left side of canvas,
+        // continuing into the start of the chromosome on the right side of canvas
+        const chromosomeMidpoint = minimalProps.length / 2;
+        const props = {
+          ...minimalProps,
+          viewportStart: random(chromosomeMidpoint, minimalProps.length),
+          viewportEnd: random(chromosomeMidpoint)
+        };
+        const { viewportAreas } = getRenderedViewport(props);
+        const expectedViewportAreaStyles = [
+          {
+            x: 0,
+            width: Math.round(props.viewportEnd * scalingFactor)
+          },
+          {
+            x: Math.round(props.viewportStart * scalingFactor),
+            width: Math.round(
+              (props.length - props.viewportStart) * scalingFactor
+            )
+          }
+        ];
+        assertViewportAreas(viewportAreas, expectedViewportAreaStyles);
+      });
+    });
+
+    describe('in the end of the chromosome', () => {
+      it('does not allow overlapping of brackets for small viewports', () => {
+        const smallDistance = Math.floor(
+          (2 * constants.VIEWPORT_BRACKET_BAR_WIDTH) / scalingFactor
+        );
+        const randomDistance = random(smallDistance);
+        const viewportEnd = minimalProps.length;
+        const viewportStart = viewportEnd - randomDistance;
+
+        const props = { ...minimalProps, viewportStart, viewportEnd };
+        const { openingBracket, closingBracket } = getRenderedViewport(props);
+
+        // we expect the viewport’s end to be in the end of the chromosome,
+        // and the viewport’s start to be two bracket widths to the left of viewport end
+        const expectedEndX = scalingFactor * props.length;
+        const expectedStartX =
+          expectedEndX - 2 * constants.VIEWPORT_BRACKET_BAR_WIDTH;
+        assertBracketX(openingBracket, expectedStartX);
+        assertBracketX(closingBracket, expectedEndX);
```

Comment:
What does `x` represent in here and other places?

Reply:
X-coordinate on the xy coordinate system.

Reply:
Would it be better to use `position` instead of `X` as we do not use `Y` at all?

`assertBracketX` -> `assertBracketPosition`
`expectedStartX` -> `expectedStartPosition`
`expectedLabel1X` -> `expectedLabel1Position`
`expectedPointerX` -> `expectedPointerPosition`

Reply:
🤷‍♂ 

one might argue that position implies both x and y, and what's being checked in the test is only the x-coordinate

Reply:
Would it not cause some confusion if we want to use the slider vertically?

Reply:
Not sure what you mean. If you mean that we might want to take the whole stick and draw it vertically rather than horizontally, then the current implementation won't work (unless we first draw the svg horizontally and then do css transform: rotate(90deg) on it. But drawing the stick vertically was never a requirement.